optimum/habana/transformers/trainer_seq2seq.py

```diff
@@ -285,7 +309,7 @@ def prediction_step(
         # Retrieves GenerationConfig from model.generation_config
         gen_config = self.model.generation_config
         # in case the batch is shorter than max length, the output should be padded
-        if generated_tokens.shape[-1] < gen_config.max_length:
+        if gen_config.max_length is not None and generated_tokens.shape[-1] < gen_config.max_length:
```

Comment:
Same

Reply:
Sorry for the confusion. 
Are we expecting gen_config to have max_new_tokens (or) max_length?
Current patch works with max_new_tokens and if we remove this then the gen_config won't have max_length and generated_tokens.shape[-1] < gen_config.max_length fails due to NoneType comparison.

If we are expecting max_length to be in gen_config, then changes need be made to make the tensors static according to max_length.